substrate/frame/multisig/src/lib.rs

```diff
@@ -109,7 +124,9 @@ where
 	MaxApprovals: Get<u32>,
 {
 	/// The extrinsic when the multisig operation was opened.
-	when: Timepoint<BlockNumber>,
+	/// `timepoint` is an optional extra security measure, which can be enabled by the creator of
+	/// the multisig during the creation by supplying a `timepoint` for the `maybe_when` field.
```

Comment:
Personally i think that the level of security should be decided upon by the approvers - not the creator.  
Now the creator can weaken the security of the other participants by using `None`.  

When the timepoint is always present, then the approvers can decide ad-hoc if they want to approve just this single call by sending `Some(_)`, or if they want to also approve all future versions of this by using `None`.  
This is very similar to immortal transactions.

Reply:
As a UI dev using these pallets a lot, this sounds like worsening the DX:
- I need to understand all this, when I didn't before
- I need to take a decision for my users whether to pass the time-point or not, or pass this complexity down to my users by having a checkbox, and an explanation.
- the default, and easiest route, not caring about this, is not the safest unfortunately

Zooming out, this PR was supposed to make DX better. My opinion is that it's not the case. As an implementer, I used to send `None` with the first tx, and managed to find the timepoint for the subsequent approvals, it 1 call to make. It was actually easier than having to think about all this IMHO.

Reply:
@Tbaut, valid points, I mostly agree, but mostly. 

Here is my take (from the issue #2541):
>
>1. providing the timepoint is an arduous task. Let me elaborate: 
>    - no function is exposed in the library for finding the `timepoint` of a multisig operation. 
>    - see this example for how many steps are required for finding the `timepoint`: https://github.com/paritytech/txwrapper-core/tree/768bb445beb2907582b2d5e13ade3be5d995af3e/packages/txwrapper-examples/multisig#timepoint-of-approveasmulti--storage
>    - this complexity is probably discouraging developers from using multisig pallet and substrate
>2. it is not obvious what `timepoint`s purpose is in the pallet. It is confusing for most of the developers to encounter `timepoint`

In short, with the current state of this PR, we achieve the following:
- purpose of the `timepoint` is clearly explained.
- BIG WIN: approving a multisig way easier (due to `timepoint` is not a necessity anymore for the most use cases)

--- 

As to your points:
> I need to understand all this, when I didn't before.

I'd argue against you have to understand more. This PR does not introduce a new concept. `Timepoint` was already there, and the purpose of the `timepoint` has not changed. We are just making it optional because for the most multisigs, timepoint is an overkill. In other words, this PR just makes the purpose of everything clear imo.

> I need to take a decision for my users whether to pass the time-point or not, or pass this complexity down to my users by having a checkbox, and an explanation -> worse UX.

Yes, you need to. But examples and reasonings behind those are explained clearly in the docs. I'd again argue against worse UX, because previously we still had to provide timepoint, and the reason for that was not clear. The explanation for the `timepoint` should have been there in the UX in the first place imo.

> the default, and easiest route, not caring about this, is not the safest unfortunately

Yes, if anyone has a suggestion on the defaults, I'm all ears.

> Zooming out, this PR was supposed to make DX better. 

I believe it still does, a lot!

> As an implementer, I used to send None with the first tx, and managed to find the timepoint for the subsequent approvals, it's 1 call to make. It was actually easier than having to think about all this IMHO.

If you don't care about the `timepoint`, you can still send `None` and everybody else will send `None`. Previously, this was not possible, even for the multisigs that do not need `timepoint`, every approver had to provide `timepoint` and finding the `timepoint` is not easy as I explained above from the perspective of the approvers.

---

From my perspective, when zoomed out:
- the functionality of this pallet is much more clear.
- for the majority of the use cases, approvers will have a much better time.

Reply:
I should probably have said it from the beginning. I think this first assumption, that it's hard to get the timepoint of a multisig tx, is IMHO, wrong. Maybe because of the api you're using. It's literally one state call, that you must do in any case to know the currently pending multisig txs. Then the info is in the storage, here's an example of a pending multisig tx:
![image](https://github.com/paritytech/polkadot-sdk/assets/33178835/2df08012-86ac-4a56-a631-fb4269c36293)

> Yes, you need to. [decide or pass down this complexity to users]

Let's not confuse things. DX is my experience implementing this, UX is the experience of users of my application. I want to build applications that have good UX, and I don't want to feature creep them by adding options that users have no idea about. I should take the best decision for them. If the best decision is the safest decision at no additional cost for my users, and a minimal cost for me (my DX, my time) then I should include the timepoint... because.. it isn't hard IMHO.

Now I'm not willing to die on this hill, I just left this comment to give my opinion. To me it's a feature creep. I've implemented this in JS, there's been millions of hard things, retrieving the timepoint wasn't one of them. It's `api.query.multisig.multisigs.entries(address)` -> iterate over the stroage, then `storage[1].toJson().when`. I may be missing something, it may be hard in other languages, but it would mean that getting the pending multisig txs is also hard to start with.

Reply:
I understand your concerns. I just wanted to fully explain my reasoning above, that's all!
Thanks for the conversation, appreciated 👍 🙏 

This PR is old, and the related issue was opened when I first tried to understand the `multisig` pallet back then. I still think `timepoint` as is, will be confusing for newcomers. And maybe I don't know how to do it, but from what I recall, one cannot easily approve a multisig in a browser (polkadotjs app). As you mentioned, they have to execute some code afaik:

> It's api.query.multisig.multisigs.entries(address) -> iterate over the stroage, then storage[1].toJson().when.

And I just think this requirement shouldn't be there in the first place for the majority of the multisigs (or at least, there should be an explanation on the UI side on why timepoint is required).

But after all, these are my subjective takes :)

Reply:
> one cannot easily approve a multisig in a browser (polkadotjs app)

You're mistaking. UIs have all they need to find and fill the timepoint. If Pjs apps isn't good at something, it's not with the timepoint, it's with the `callData`. Which is a whole other beast. And this is worked around by many UIs including one I build, Multix, [more info about this workaround here](https://github.com/ChainSafe/Multix/wiki/Multix-integration). Now this issue is the subject of an RFC: https://github.com/polkadot-fellows/RFCs/pull/74